templates/aws-stack.yml

```diff
@@ -861,6 +868,16 @@ Resources:
                  - !Ref 'AWS::NoValue'
           - !Ref 'AWS::NoValue'
       Policies:
+        - !If
+          - AddECRPullThrough
+          - PolicyName: ECRPullThrough
+            PolicyDocument:
+              Version: '2012-10-17'
+              Statement:
+                - Effect: Allow
+                  Action:
+                    - ecr:BatchImportUpstreamImage
```

Comment:
you may also want `ecr:CreateRepository` here. That permission is required for the first pull of a new image that's never been cached before. After that, `ecr:BatchImportUpstreamImage` is enough to pull new tags of the image into the repository.